pkg/checker/interfaces.go

```diff
@@ -23,25 +23,17 @@ func InterfaceExists(ifname string) Checker {
 	}
 }
 
-func ethStats(ifname string) (*ethtool.EthtoolCmd, error) {
-	cmd := ethtool.EthtoolCmd{}
-	_, err := cmd.CmdGet(ifname)
-	if err != nil {
-		return nil, err
-	}
-	return &cmd, nil
-}
-
 // LinkSpeed checks the link speed, and complains if smaller than `min`
 // megabit/s.
-func LinkSpeed(ifname string, minSpeed int) Checker {
+func LinkSpeed(ifname string, minSpeed int64) Checker {
 	return func() error {
-		eth, err := ethStats(ifname)
+		ec := ethtool.EthtoolCmd{}
+		speed, err := ec.CmdGet(ifname)
 		if err != nil {
 			return err
 		}
-		if int(eth.Speed) < minSpeed {
-			return fmt.Errorf("link speed %d < %d", eth.Speed, minSpeed)
+		if int64(speed) < minSpeed {
```

Comment:
it's all `int64` now, and the speed is fetched directly with `CmdGet` instead of reading the lower 16 bits as previously

Reply:
It looks like `CmdGet` returns a `uint32`. Perhaps it makes more sense to use a `uint32` throughout so this function isn't advertising a larger type than what is supported? Or perhaps even create a new `type SpeedMbitps uint32`.

Reply:
good idea, I'll make a type alias

Reply:
on a second thought, given that this is used only internally in that function, I'll use `uint32` instead of a type alias. Let me know if you have a strong preference to do differently

Reply:
pushed update, PTAL